nixos/modules/services/networking/nbd.nix

```diff
@@ -0,0 +1,87 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.nbd;
+  serverConfig = pkgs.writeText "nbd-server-config" (''
+    [generic]
+    user = root
+    group = root
+    allowlist = false
+    listenaddr = ${cfg.server.listenAddress}
+    port = ${toString cfg.server.listenPort}
+  '' + concatStringsSep "\n"
+    (mapAttrsToList
+      (name: { device, authSubnet }:
+        ''
+          [${name}]
+          exportname = ${device}
+          authfile = ${pkgs.writeText "authfile" ''
+            ${authSubnet}
+          ''}
+          flush = true
+          fua = true
+        '')
+      cfg.server.exports));
+in
+{
+  imports = [ ];
+
+  options = {
+    services.nbd = {
+      server = {
+        enable = mkEnableOption "the Network Block Device (nbd) server.";
+
+        listenPort = mkOption {
+          type = types.port;
+          default = 10809;
+          description = "Port to listen on. The port is NOT automatically opened in the firewall.";
+        };
+
+        exports = mkOption {
+          description = "Block devices to make available over the network.";
+          default = { };
+          type = with types; attrsOf
+            (submodule {
+              options = {
+                device = mkOption {
+                  type = types.str;
+                  description = "Block device to export.";
```

Comment:
it'd be nice to note here that plain files are also allowed

Reply:
Done.  I also renamed `device` to `path`.